lib/cc/analyzer/issue_relative_path_validation.rb

```diff
@@ -1,16 +1,32 @@
+require "pathname"
+
 module CC
   module Analyzer
     class IssueRelativePathValidation < Validation
       def valid?
-        path && !path.start_with?("/")
+        path &&
+          !path.start_with?("/") &&
+          relative_to?(MountedPath.code.container_path)
       end
 
       def message
-        "Path must be relative"
+        "Path must be relative to the project directory"
       end
 
       private
 
+      def relative_to?(directory)
+        expanded_base = expand_path(directory)
+        expanded_path = expand_path(path)
+
+        expanded_path.start_with?(expanded_base)
+      end
+
+      def expand_path(path)
+        cleaned = Pathname.new(path).cleanpath.to_s
```

Comment:
Any reason not to use `Pathname#expand_path`? I think it pretty much would basically replace this whole method from what i've tested of its behavior.